compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2177,61 +2177,47 @@ impl<'tcx> FmtPrinter<'_, 'tcx> {
         define_scoped_cx!(self);
 
         let mut region_index = self.region_index;
+        let mut next_name = |this: &Self| loop {
+            let name = name_by_region_index(region_index);
+            region_index += 1;
+            if !this.used_region_names.contains(&name) {
+                break name;
+            }
+        };
+
         // If we want to print verbosely, then print *all* binders, even if they
         // aren't named. Eventually, we might just want this as the default, but
         // this is not *quite* right and changes the ordering of some output
         // anyways.
         let (new_value, map) = if self.tcx().sess.verbose() {
             // anon index + 1 (BrEnv takes 0) -> name
-            let mut region_map: BTreeMap<u32, Symbol> = BTreeMap::default();
+            let mut region_map: FxHashMap<_, _> = Default::default();
```

Comment:
This code is a bit sus to me. This relies on us *not* having duplicate `BoundRegionKind`s in binders. This is how the compiler behaves today (or should barring bugs), but there isn't nothing to say that we *can't* (you could, for example, imagine that we drop the anon index).

This makes more sense as just a `Vec` where each index is just the `BoundRegionKind` that corresponds to the binder (or just some placeholder value for non-regions) and then use `BoundRegion.var.as_usize()` as an index into that vec, which is exactly how bound variables work.

(I'm making these changes currently for the bug below).